maptemplate.c

```diff
@@ -1310,6 +1316,15 @@ static int processItemTag(layerObj *layer, char **line, shapeObj *shape)
       tagValue = msReplaceSubstring(tagValue, "$value", itemValue);
       msFree(itemValue);
 
+      if (padding != -1) {
+          int paddedSize = sizeof(tagValue) + (padding * sizeof(char*));
```

Comment:
several errors:
- sizeof(tagValue) is sizeof(char*) = 8 on a 64 bit host. You probably meant strlen(tagValue)
- sizeof(char*) should likely be sizeof(char) (or can be just omitted since it is guaranteed that sizeof(char)==1)
- once the above is fixed, you probably lack a +1 for the final zero terminator (for snprintf() purposes), and thus remove the +1 in the msSmallMalloc()